### PR TITLE
Output searched locations on import failure.

### PR DESCRIPTION
**Pull request by chriseth**
Revival of https://github.com/ethereum/solidity/pull/12267

### CODE REVIEW

**Review by cameel — COMMENTED** [review]
CLI tests need updating.

Also, I think it would be a good idea to also print allowed paths in the same way when you get a message that your path is not allowed.

**Review by ekpyron — DISMISSED** [review]

**Comment by ekpyron** [comment]
Needs CLI test updates.

File: test/solc/CommandLineInterfaceAllowPaths.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ ImportCheck checkImport(
 		return ImportCheck::OK();
 
 	static regex const sourceNotFoundErrorRegex{
-		R"(^Error \(6275\): Source ".+" not found: (.*)\.\n)"
+		R"(^Error \(6275\): Source "[^"]+" not found: (.*)\.\n)"
```
Comment (ekpyron) [inline test/solc/CommandLineInterfaceAllowPaths.cpp:103]: I wonder if we should expect that somebody may be parsing this just like we did here and therefore at least generously add a changelog entry for this...

**Review by ekpyron — APPROVED** [review]

File: test/solc/CommandLineInterfaceAllowPaths.cpp (hunk 1 above)
Comment (chriseth) [reply test/solc/CommandLineInterfaceAllowPaths.cpp:103]: The problem was the greedy match to begin with.

**Pull request merged**